invoices/invoiceregistry.go

```diff
@@ -611,6 +616,57 @@ func (i *InvoiceRegistry) cancelSingleHtlc(hash lntypes.Hash,
 	return nil
 }
 
+// processKeySend just-in-time inserts an invoice if this htlc is a key send
+// htlc.
+func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx,
+	hash lntypes.Hash) error {
+
+	// Retrieve key send record if present.
+	preimageSlice, ok := ctx.customRecords[record.KeySend]
```

Comment:
I considered that, but didn't want to do the look up twice. Would you, with the external check, pass in the preimage slice? Also I tried to keep all the key send functionality together in this function. But I have no problem with changing it either.

Reply:
agreed, the double lookup is def undesired, though it can be avoided by passing the slice as you described. for now i think it's fine to keep it this way, esp considering that the locking is more complicated since we need to JIT insert an invoice. perhaps just something to note on the road to a unified keysend via AMP.